specs/_features/peerdas/p2p-interface.md

```diff
@@ -0,0 +1,180 @@
+| `KZG_COMMITMENTS_INCLUSION_PROOF_DEPTH`   | `uint64(floorlog2(get_generalized_index(BeaconBlockBody, 'blob_kzg_commitments')))` (= 4) | <!-- predefined --> Merkle proof index for `blob_kzg_commitments` |
+
+### Configuration
+
+| Name                                     | Value                             | Description                                                         |
+|------------------------------------------|-----------------------------------|---------------------------------------------------------------------|
+| `DATA_COLUMN_SIDECAR_SUBNET_COUNT`              | `32`                               | The number of data column sidecar subnets used in the gossipsub protocol.  |
+
+### Containers
+
+#### `DataColumnSidecar`
+
+```python
+class DataColumnSidecar(Container):
+    index: LineIndex  # Index of column in extended matrix
+    column: DataColumn
+    kzg_commitments: List[KZGCommitment, MAX_BLOB_COMMITMENTS_PER_BLOCK]
+    kzg_proofs: List[KZGProof, MAX_BLOB_COMMITMENTS_PER_BLOCK]
+    signed_block_header: SignedBeaconBlockHeader
+    kzg_commitments_inclusion_proof: Vector[Bytes32, KZG_COMMITMENTS_INCLUSION_PROOF_DEPTH]
+```
+
+#### `DataColumnIdentifier`
+
+```python
+class DataColumnIdentifier(Container):
+    block_root: Root
+    index: LineIndex
+```
+
+### Helpers
+
+##### `verify_data_column_sidecar_kzg_proof`
+
+```python
+def verify_data_column_sidecar_kzg_proof(sidecar: DataColumnSidecar) -> bool:
+    """
+    Verify if the proofs are correct
+    """
+    row_ids = [LineIndex(i) for i in range(len(sidecar.column))]
+    assert len(sidecar.column) == len(sidecar.kzg_commitments) == len(sidecar.kzg_proofs)
+
+    # KZG batch verifies that the cells match the corresponding commitments and proofs
+    return verify_cell_proof_batch(
+        row_commitments=sidecar.kzg_commitments,
+        row_ids=row_ids,  # all rows
+        column_ids=[sidecar.index],
+        datas=sidecar.column,
+        proofs=sidecar.kzg_proofs,
+    )
+```
+
+##### `verify_data_column_sidecar_inclusion_proof`
+
+```python
+def verify_data_column_sidecar_inclusion_proof(sidecar: DataColumnSidecar) -> bool:
+    """
+    Verify if the given KZG commitments included in the given beacon block.
+    """
+    gindex = get_subtree_index(get_generalized_index(BeaconBlockBody, 'blob_kzg_commitments'))
+    return is_valid_merkle_branch(
+        leaf=hash_tree_root(sidecar.kzg_commitments),
+        branch=sidecar.kzg_commitments_inclusion_proof,
+        depth=KZG_COMMITMENTS_INCLUSION_PROOF_DEPTH,
+        index=gindex,
+        root=sidecar.signed_block_header.message.body_root,
+    )
+```
+
+##### `compute_subnet_for_data_column_sidecar`
+
+```python
+def compute_subnet_for_data_column_sidecar(column_index: LineIndex) -> SubnetID:
+    return SubnetID(column_index % DATA_COLUMN_SIDECAR_SUBNET_COUNT)
+```
+
+
+### The gossip domain: gossipsub
```

Comment:
Should add deprecation of topic `blob_sidecar_{subnet_id}`

Reply:
+1

Reply:
I think blob sidecar subnets could be a backup channel for data distribution during the fork transition. Maybe let's revisit it after evaluate the performance and network stability in prototyping?

Reply:
but the nodes wouldn't have full blobs (commented the same elsewhere as well)

Reply:
addressed in https://github.com/ethereum/consensus-specs/pull/3574/commits/d23452d02bb54e9bffc5547fde00cfd1537d6d42